### PR TITLE
Query: inline deterministic fields

### LINKED ISSUE #10547: Query: Inline deterministic static member access in SQL
```C#
using System;
using System.Linq;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace EFSampleApp
{
    public class Program
    {
        public static void Main(string[] args)
        {
            using (var db = new MyContext())
            {
                // Recreate database
                db.Database.EnsureDeleted();
                db.Database.EnsureCreated();

                // Seed database


                db.SaveChanges();
            }

            using (var db = new MyContext())
            {
                // Run queries
                var query = db.Blogs.Where(b => b.Title == string.Empty).ToList();
            }

            Console.WriteLine("Program finished.");
        }
    }


    public class MyContext : DbContext
    {
        private static ILoggerFactory LoggerFactory => new LoggerFactory().AddConsole(LogLevel.Trace);

        // Declare DBSets
        public DbSet<Blog> Blogs { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            // Select 1 provider
            optionsBuilder
                .UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=_ModelApp;Trusted_Connection=True;Connect Timeout=5;ConnectRetryCount=0")
                //.UseSqlite("filename=_modelApp.db")
                //.UseInMemoryDatabase(databaseName: "_modelApp")
                .EnableSensitiveDataLogging()
                .UseLoggerFactory(LoggerFactory);
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            // Configure model
        }
    }

    public class Blog
    {
        public int Id { get; set; }
        public string Title { get; set; }
    }
}
```

EF Core version 2.0.1

`string.Empty` is static member whose value cannot be changed. So above query cannot be reused for different thing. Since `string.Empty` is member access we created parameter for it. We should avoid creating parameter for static members.

### PR DESCRIPTION
**Pull request by SSkovboiSS**
Fixes #10547

### CODE REVIEW

**Comment by ralmsdeveloper** [comment]
I tested it and it also solves: #10563

**Comment by SSkovboiSS** [comment]
@ralmsdeveloper I supposed that, but unfortunately couldn't get oracle tests to run.

They throw `AggregateException` with message TNS:could not resolve the connect identifier specified.
Probably I need to install some kind of SDK for Oracle to make this work.

**Comment by ralmsdeveloper** [comment]
You have to change the connection of the OracleTestStore.cs file, also follow the steps of:
[https://github.com/aspnet/EntityFrameworkCore/tree/dev/samples/OracleProvider](https://github.com/aspnet/EntityFrameworkCore/tree/dev/samples/OracleProvider)

NOTE: Have a test branch to use this connection.

File: src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs [hunk 1]
```diff
@@ -219,7 +219,9 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
         /// </summary>
         protected override Expression VisitMember(MemberExpression memberExpression)
         {
-            if (!_partialEvaluationInfo.IsEvaluatableExpression(memberExpression))
+            if (memberExpression.Member is FieldInfo fieldInfo
+                && (fieldInfo.IsStatic && fieldInfo.IsInitOnly || fieldInfo.IsLiteral)
+                || !_partialEvaluationInfo.IsEvaluatableExpression(memberExpression))
```
Comment (smitpatel) [inline src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: Remove IsLiteral. Unless it is absolutely necessary in which case add a regression test.

Comment (SSkovboiSS) [reply src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: Removed.

File: test/EFCore.SqlServer.FunctionalTests/Query/SimpleQuerySqlServerTest.Functions.cs [hunk 2]
```diff
@@ -1063,12 +1063,7 @@ public override void Indexof_with_emptystring()
             base.Indexof_with_emptystring();
 
             AssertSql(
-                @"@__Empty_0='' (Size = 4000)
-
-SELECT CASE
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/Query/SimpleQuerySqlServerTest.Functions.cs]: I added a commit in dev which would stop this adverse optimization to run. Can you rebase on dev? This test may require an update.

Comment (SSkovboiSS) [reply test/EFCore.SqlServer.FunctionalTests/Query/SimpleQuerySqlServerTest.Functions.cs]: Yes, I'm doing rebase and will verify if this test passes.

**Comment by SSkovboiSS** [comment]
It is ready for merge I suppose.

**Comment by smitpatel** [comment]
@SSkovboiSS - Thanks for contribution. Sorry for delay in merge, I had to manually verify that this change does not break Oracle tests since public CI servers don't run them.

**Pull request merged**